tensorboard/webapp/plugins/plugins_container_test.ts

```diff
@@ -467,6 +508,54 @@ describe('plugins_component', () => {
       ).not.toBeNull();
     });
 
+    describe('custom error templates', () => {
+      it('shows warning when environment failed NOT_FOUND', () => {
+        store.overrideSelector(getActivePlugin, null);
+        store.overrideSelector(getPluginsListLoaded, {
+          state: DataLoadState.FAILED,
+          lastLoadedTimeInMs: null,
+          failureCode: PluginsListFailureCode.NOT_FOUND,
+        });
+        const fixture = TestBed.createComponent(
+          CustomizedErrorTemplatesComponent
+        );
+        fixture.detectChanges();
+
+        expect(
+          fixture.debugElement.query(By.css('.environment-not-loaded'))
+        ).toBeNull();
+        expect(
+          fixture.debugElement.query(By.css('.custom-not-found-template'))
+        ).not.toBeNull();
+        expect(
+          fixture.debugElement.query(By.css('.custom-unknown-template'))
+        ).toBeNull();
+      });
+
+      it('shows warning when environment failed UNKNOWN', () => {
+        store.overrideSelector(getActivePlugin, null);
+        store.overrideSelector(getPluginsListLoaded, {
+          state: DataLoadState.FAILED,
+          lastLoadedTimeInMs: null,
+          failureCode: PluginsListFailureCode.UNKNOWN,
+        });
+        const fixture = TestBed.createComponent(
+          CustomizedErrorTemplatesComponent
+        );
+        fixture.detectChanges();
+
+        expect(
+          fixture.debugElement.query(By.css('.environment-not-loaded'))
+        ).toBeNull();
+        expect(
+          fixture.debugElement.query(By.css('.custom-not-found-template'))
+        ).toBeNull();
+        expect(
+          fixture.debugElement.query(By.css('.custom-unknown-template'))
```

Comment:
can we do a basic content sanity check instead? If we rely on css, the test behavior may change when I add a CSS/class line to the template.

Reply:
Done.